oak_restricted_kernel/src/interrupts.rs

```diff
@@ -164,13 +164,164 @@ mutable_interrupt_handler_with_error_code!(
+    error!("KERNEL PANIC: INVALID OPCODE!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn device_not_available_handler(stack_frame: InterruptStackFrame) {
+    error!("KERNEL PANIC: DEVICE NOT AVAILABLE!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn invalid_tss_handler(stack_frame: InterruptStackFrame, error_code: u64) {
+    error!("KERNEL PANIC: INVALID TSS!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    error!("Error code: {:?}", error_code);
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn segment_not_present_handler(
+    stack_frame: InterruptStackFrame,
+    error_code: u64,
+) {
+    error!("KERNEL PANIC: SEGMENT NOT PRESENT!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    error!("Error code: {:?}", error_code);
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn stack_exception_handler(
+    stack_frame: InterruptStackFrame,
+    error_code: u64,
+) {
+    error!("KERNEL PANIC: STACK EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    error!("Error code: {:?}", error_code);
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn x87_floating_point_handler(stack_frame: InterruptStackFrame) {
+    error!("KERNEL PANIC: X87 FLOATING POINT EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn alignment_check_handler(
+    stack_frame: InterruptStackFrame,
+    error_code: u64,
+) {
+    error!("KERNEL PANIC: ALIGNMENT CHECK EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    error!("Error code: {:?}", error_code);
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn machine_check_handler(stack_frame: InterruptStackFrame) -> ! {
+    error!("KERNEL PANIC: MACHINE CHECK EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
+extern "x86-interrupt" fn simd_fp_handler(stack_frame: InterruptStackFrame) {
+    error!("KERNEL PANIC: SIMD FLOATING POINT EXCEPTION!");
+    error!(
+        "Instruction pointer: {:#016x}",
+        stack_frame.deref().instruction_pointer.as_u64()
+    );
+    shutdown::shutdown();
+}
+
 pub fn init_idt() {
     let mut idt = InterruptDescriptorTable::new();
+    idt.divide_error.set_handler_fn(divide_error_handler); // vector 0
```

Comment:
Could you please add a link where this list of interrupts was taken from?

Reply:
I added a note for the AMD manual as I had that open; Intel has its own list.